utilities/hooks/useForm.ts

```diff
@@ -0,0 +1,40 @@
+import { ChangeEventHandler, useReducer } from 'react'
+import { FormProps } from '../types/formTypes'
+interface ActionState {
+    type: string
+    payload?: any
+}
+function reducer(state: FormProps, action: ActionState) {
+    switch (action.type) {
+        case 'handle_change': {
+            const { name, value } = action.payload
+            return { ...state, [name]: value }
+        }
+        case 'handle_reset': {
+            return state
```

Comment:
Resetting should always do a full reset. Returning `state` will not actually reset, itll just give you back the current state with it's values. return initial state